packages/remix/src/utils/utils.ts

```diff
@@ -32,14 +36,19 @@ export function assertValidClerkState(val: any): asserts val is ClerkState | und
  * @returns string
  */
 export const getEnvVariable = (name: string, context: AppLoadContext | undefined): string => {
+  // Remix + Cloudflare pages
+  if (typeof (context?.cloudflare as CloudflareEnv)?.env !== 'undefined') {
+    return (context?.cloudflare as CloudflareEnv).env[name] || '';
+  }
+
   // Node envs
   if (typeof process !== 'undefined') {
     return (process.env && process.env[name]) || '';
   }
 
   // Cloudflare pages
   if (typeof context !== 'undefined') {
-    const contextEnv = context?.env as Record<string, string>;
+    const contextEnv = (context as CloudflareEnv)?.env;
```

Comment:
We can change to the following line to:
```suggestion
    const baseContext = (context?.cloudflare || context) as Record<string, unknown>;
    const contextEnv = (baseContext?.env || baseContext?.env) as Record<string, string>;
```